File: frontend/src/types/project.ts
/** 项目状态 */
export type ProjectStatus = 'planning' | 'in_progress' | 'closed'

/** 项目信息 */
export interface Project {
  id: number
  projectCode: string
  name: string
  description?: string
  type: string
  status: ProjectStatus
  promotionEnabled?: boolean
  startDate?: string
  endDate?: string
  orgId?: number
  projectManagerId?: number
  createdAt: string
}

/** 项目创建请求 */
export interface ProjectCreateRequest {
  projectCode?: string
  name: string
  description?: string
  type: string
  promotionEnabled?: boolean
  startDate?: string
  endDate?: string
  orgId?: number
}

/** 项目更新请求 */
export interface ProjectUpdateRequest {
  name?: string
  description?: string
  type?: string
  status?: ProjectStatus
  startDate?: string
  endDate?: string
}

/** 项目成员 */
export interface ProjectMember {
  id: number
  projectId: number
  userId: number
  role: string
  promotionUnitId?: number
  joinedAt: string
}

File: backend/pm-project/src/main/java/com/pm/project/domain/Project.java
package com.pm.project.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import com.pm.common.base.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

/**
 * 项目实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("project")
@Schema(description = "项目")
public class Project extends BaseEntity {

    @Schema(description = "项目编码")
    private String projectCode;

    @Schema(description = "项目名称")
    private String name;

    @Schema(description = "项目描述")
    private String description;

    @Schema(description = "项目类型：software_dev/system_integration/ops")
    private String type;

    @Schema(description = "项目状态：planning/in_progress/closed")
    private String status;

    @Schema(description = "是否启用集团推广模块")
    private Boolean promotionEnabled;

    @Schema(description = "计划开始日期")
    private LocalDate startDate;

    @Schema(description = "计划结束日期")
    private LocalDate endDate;

    @Schema(description = "所属组织ID")
    private Long orgId;

    @Schema(description = "创建人ID")
    private Long ownerUserId;
}
